CIScripts/Test/Tests/vRouterAgent.Tests.ps1

```diff
@@ -0,0 +1,131 @@
+Param (
+    [Parameter(Mandatory=$true)] [string] $TestbedAddr
+)
+
+. $PSScriptRoot\..\Utils\CommonTestCode.ps1
+. $PSScriptRoot\..\TestConfigurationUtils.ps1
+. $PSScriptRoot\..\..\Common\Aliases.ps1
+. $PSScriptRoot\..\..\Common\VMUtils.ps1
+
+. $PSScriptRoot\..\GetTestConfigurationCodiLegacy.ps1
+$TestConf = Get-TestConfiguration
+$Session = New-PSSession -ComputerName $TestbedAddr -Credential (Get-VMCreds)
+
+Describe "vRouter Agent" {
+
+    Context "installing using MSI" {
+        It "registers Agent service" {
+            Assert-IsAgentServiceRegistered -Session $Session
+        }
+
+        It "does not enable Agent service" {
+            Assert-IsAgentServiceDisabled -Session $Session
+        }
+    }
+
+    Context "uninstalling using MSI" {
+        It "unregisters Agent service" {
+            Assert-IsAgentServiceUnregistered -Session $Session
+        }
+
+        BeforeEach {
+            Assert-IsAgentServiceRegistered -Session $Session
+            Uninstall-Agent -Session $Session
```

Comment:
How this interacts with `Uninstall-Agent` that's in global `AfterEach`? (I guess second uninstallation just fails silently (as we don't check exitcode of msiexec). Do we want to keep it that way?